dolphinscheduler-task-plugin/dolphinscheduler-task-datafactory/src/main/java/org/apache/dolphinscheduler/plugin/task/datafactory/DatafactoryHook.java

```diff
@@ -0,0 +1,147 @@
+                .clientId(AZURE_CLIENT_ID)
+                .clientSecret(AZURE_CLIENT_SECRET)
+                .tenantId(AZURE_SECRET_TENANT_ID)
+                .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
+                .build();
+        return DataFactoryManager.authenticate(credential, profile);
+    }
+
+    public Boolean startDatafactoryTask(DatafactoryParameters parameters) {
+        logger.info("initDatafactoryTask ......");
+        PipelineResource pipelineResource = getPipelineResource(parameters);
+        if (pipelineResource == null) {
+            return false;
+        }
+        logger.info("startDatafactoryTask ......");
+        CreateRunResponse run = pipelineResource.createRun();
+        if (StringUtils.isEmpty(run.runId())) {
+            return false;
+        }
+        runId = run.runId();
+        parameters.setRunId(runId);
+        return true;
+    }
+
+    public Boolean cancelDatafactoryTask(DatafactoryParameters parameters) {
+        logger.info("cancelTask ......");
+        PipelineRuns pipelineRuns = client.pipelineRuns();
+        try {
+            pipelineRuns.cancel(parameters.getResourceGroupName(), parameters.getFactoryName(), runId);
+        } catch (RuntimeException e) {
+            logger.error("failed to cancel datafactory task: " + e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+    public DatafactoryStatus queryDatafactoryTaskStatus(DatafactoryParameters parameters) {
+        logger.info("queryDatafactoryTaskStatus ......");
+
+        PipelineRuns pipelineRuns = client.pipelineRuns();
+        PipelineRun pipelineRun =
+                pipelineRuns.get(parameters.getResourceGroupName(), parameters.getFactoryName(), runId);
+
+        if (pipelineRun != null) {
+            logger.info("queryDatafactoryTaskStatus ......{}", pipelineRun.status());
+            return DatafactoryStatus.valueOf(pipelineRun.status());
+        }
+        return null;
+    }
+
+    private PipelineResource getPipelineResource(DatafactoryParameters parameters) {
+        return client.pipelines().get(parameters.getResourceGroupName(), parameters.getFactoryName(),
+                parameters.getPipelineName());
+    }
+
+    @SneakyThrows
+    public Boolean queryStatus(DatafactoryParameters parameters) {
+        List<DatafactoryStatus> stopStatusSet = Arrays.asList(taskFinishFlags);
+        int maxRetry = 5;
+        while (maxRetry > 0) {
+            DatafactoryStatus status = queryDatafactoryTaskStatus(parameters);
+
+            if (status == null) {
+                maxRetry--;
+                continue;
+            }
+
+            if (stopStatusSet.contains(status)) {
+                if (status.equals(DatafactoryStatus.Succeeded)) {
+                    return true;
+                }
+                return false;
+            }
+            logger.debug("wait 10s to recheck finish status....");
+            Thread.sleep(10000);
```

Comment:
Is it better to make this 10000 configurable so that users can adjust under extreme network conditions?